parsl/monitoring/radios/udp.py

```diff
@@ -23,25 +25,37 @@ def __init__(self, *, port: Optional[int] = None, atexit_timeout: int = 3, addre
 
     def create_sender(self) -> MonitoringRadioSender:
         assert self.port is not None, "self.port should have been initialized by create_receiver"
-        return UDPRadioSender(self.address, self.port)
+        return UDPRadioSender(self.address, self.port, self.hmac_key)
 
     def create_receiver(self, run_dir: str, resource_msgs: Queue) -> MonitoringRadioReceiver:
+        # 128 byte key length is chosen based on Microsoft's recommendation
+        # here:
+        # https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.hmacsha512.-ctor
+        # and on comments in RFC 2104 section 2 that the key length be at
+        # least as long as the hash output (64 bytes in the case of SHA512).
+        # RFC 2014 section 3 talks about periodic key refreshing. This key is
+        # not refreshed inside a workflow run, but each separate workflow run
+        # uses a new key.
+        self.hmac_key = secrets.token_bytes(128)
+
```

Comment:
From a single-point-of-authority standpoint, what about defining the algorithm on the instance?  I'm not suggesting to allow it be specified (but also a possibility), but define it once, here, and then similarly pass it through the start-up chain to `udp_router_starter()`.  Perhaps:

```python
self.hmac_key = secrets.token_bytes(128)
self.digest_function = 'sha512'

udp_receiver = start_udp_receiver(..., hmac_digest=self.digest_function, ...)
```

Then the `origin_hmac = ` and `recomputed_hmac =` lines would not have a hard-coded-and-lucky-we're-looking-now setup.

Reply:
dc608c3dbb adds a new config value for the digest function. It wasn't much extra to expose that through the initializer.